### ISSUE TITLE
False negative when ParamSpec appears multiple times in a signature and is assigned incompatible signatures

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3350]
This should result in an error but does not.

```python
from typing import Callable, ParamSpec

P = ParamSpec("P")

def foo(x: Callable[P, int], y: Callable[P, int]) -> Callable[P, bool]:
    ...

def a(p0: int) -> int:
    ...

def b(p0: int, p2: str) -> int:
    ...

f1 = foo(a, b)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.238, which I just published. It will also be included in the next release of pylance.
